distro/bin/avail-config

```diff
@@ -0,0 +1,204 @@
+# not already have one in place.
+if [ ! -e "$CONFIG/logging.properties" ]; then
+    cp "$AVAIL_HOME/etc/logging.properties" "$CONFIG/logging.properties"
+elif [ ! -f "$CONFIG/logging.properties" ]; then
+    echo "Fatal error: $CONFIG/logging.properties exists but is not a regular file"
+    exit 1
+fi
+
+# Create an empty renames file, but only if the user does not already have an
+# existing one.
+if [ ! -e "$CONFIG/renames" ]; then
+    touch "$CONFIG/renames"
+elif [ ! -f "$CONFIG/renames" ]; then
+    echo "Fatal error: $CONFIG/renames exists but it is not a regular file"
+    exit 1
+fi
+
+# Create the repository directory if necessary. Fix the permissions if it is
+# not readable and writable by the current user.
+REPO_DIR=$AVAIL_USER/repos
+if [ ! -e "$REPO_DIR" ]; then
+  install -d -m 700 "$REPO_DIR"
+elif [ ! -d "$REPO_DIR" ]; then
+  echo "Fatal error: $REPO_DIR exists but is not a directory."
+  exit 1
+elif [ ! -r "$REPO_DIR" -o ! -w "$REPO_DIR" ]; then
+chmod u+rw "$REPO_DIR"
+fi
+
+USE_EXAMPLE_ROOTS=true
+DRY_RUN=
+FORCE_FOREGROUND=
+ENABLE_ASSERTIONS=
+declare -a ADDITIONAL_ROOTS
+
+if [ "$SKIP_OPT" = true ]; then
+    OPTIND=1
+else
+  # Read options
+  while getopts ":a:cdeEf?" opt; do
+    case ${opt} in
+      a )
+        # [a]dd an entry to the AVAIL_ROOTS list
+        ADDITIONAL_ROOTS+=( "$OPTARG" )
+        ;;
+      c )
+        # add the [c]urrent directory to the AVAIL_ROOTS list
+        improvised_name="${PWD##*/}"
+        improvised_name="${improvised_name%%.*}"
+        improvised_name="local.$USER.${improvised_name// /}"
+        # the root's name is given as local.username.directoryname
+        ADDITIONAL_ROOTS+=( "$improvised_name=$REPO_DIR/$improvised_name.repo,$PWD" )
+        ;;
+      d )
+        # perform a [d]ry run and output the final command instead of running it
+        DRY_RUN=true
+        ;;
+      e )
+        # Enable assertions
+        ENABLE_ASSERTIONS=true
+        ;;
+      E )
+        # omit the [E]xamples in the default roots
+        USE_EXAMPLE_ROOTS=
+        ;;
+      f )
+        # run in the [f]oreground
+        FORCE_FOREGROUND=true
+        ;;
+      \? )
+        if [ "$OPTARG" != "?" ]; then
+          # getopts uses ? to indicate unknown option
+          echo "Unknown option: $OPTARG" 1>&2
+        else
+          # this is a "real" -? flag, so print the [h]elp
+          echo "Usage:" 
+          echo "  ANY -?"
+          echo "  avail-dev [-a roots] [-c] [-d] [-E] [-f] [initial_module]"
+          echo "  avail-server [-a roots] [-c] [-d] [-E] [-f]"
+          echo "  availc ..."
+          echo "Commands:"
+          echo "  avail-dev    Launch the interactive workbench UI."
+          echo "  avail-server Launch the avail web server."
+          echo "  availc       Launch the avail CLI compiler. This tool has its own command"
+          echo "               and argument syntax, viewable with the -? option."
```

Comment:
It would be nice to use the name of the command (`$0`) to decide which help information to show, but I won't hold up a merge for this.